PMM-14624: Fix theme inheritance between users

dmitri-saricev-3pillargloball opened the pull request:
Remove localStorage theme persistence and sync theme from Grafana API on login to prevent new users from inheriting previous user's theme preferences.
https://perconadev.atlassian.net/browse/PMM-14624

result:

https://github.com/user-attachments/assets/fd69c513-8a77-4663-a69c-7d97d2ca1309



Link to the Feature Build:
https://github.com/Percona-Lab/pmm-submodules/pull/4149
https://pmm.cd.percona.com/blue/organizations/jenkins/pmm3-submodules/detail/pmm3-submodules/2057/pipeline/


If this PR adds, removes or alters one or more API endpoints, please review and add or update the relevant [API documentation](https://github.com/percona/pmm/tree/v3/docs/api) as well:

- [ ] API Docs updated

If this PR is related to some other PRs in this or other repositories, please provide links to those PRs:

- Links to related pull requests (optional).

dmitri-saricev-3pillargloball committed:
PMM-14624: Fix theme inheritance between users

Remove localStorage theme persistence and sync theme from Grafana API on login to prevent new users from inheriting previous user's theme preferences.

matejkubinec commented on ui/apps/pmm/src/api/user.ts:
```diff
@@ -17,6 +17,11 @@ export const getCurrentUserOrgs = async () => {
 };
 
+export const getUserPreferences = async () => {
+  const res = await grafanaApi.get<UpdatePreferencesBody>('/user/preferences');
```
The type `UpdatePreferencesBody` isn't probably correct (if it's the same payload you can just export it like so:

```tsx
export type GetPreferencesResponse = UpdatePreferencesBody;
```

matejkubinec replied:
```suggestion
  const res = await grafanaApi.get<GetPreferencesResponse>('/user/preferences');
```

matejkubinec commented on ui/apps/pmm/src/Providers.tsx:
```diff
@@ -8,31 +8,39 @@ import { UpdatesProvider } from 'contexts/updates';
 import { Outlet } from 'react-router-dom';
+import { useThemeSync } from 'hooks/useThemeSync';
+
+const ProvidersContent: FC<PropsWithChildren> = ({ children }) => {
```
Could you rename this component to `ThemeSyncProvider` so it aligns with the usage

dmitri-saricev-3pillargloball committed:
Merge branch 'v3' into PMM-14624-Fix-theme-inheritance

dmitri-saricev-3pillargloball committed:
change provider name

matejkubinec commented on ui/apps/pmm/src/api/user.ts:
```diff
@@ -7,6 +7,8 @@ import {
 } from 'types/user.types';
 import { api, grafanaApi } from './api';
 
+export type GetPreferenceResponse = UpdatePreferencesBody;
```
Should be in `user.types`

matejkubinec commented on ui/apps/pmm/src/hooks/api/useUser.ts:
```diff
@@ -60,6 +61,15 @@ export const useCurrentUserOrgs = (
   });
 
+export const useUserPreferences = (
+  options?: Partial<UseQueryOptions<UpdatePreferencesBody>>
```
Should also use `GetPreferenceResponse` from `types/user.types`

matejkubinec commented on ui/apps/pmm/src/Providers.tsx:
```diff
@@ -8,31 +8,39 @@ import { UpdatesProvider } from 'contexts/updates';
 import { Outlet } from 'react-router-dom';
+import { useThemeSync } from 'hooks/useThemeSync';
+
+const ThemedyProvider: FC<PropsWithChildren> = ({ children }) => {
```
Typo?

dmitri-saricev-3pillargloball replied:
yes ))) sorry

dmitri-saricev-3pillargloball committed:
Fix PR review comments

- Move GetPreferenceResponse type to user.types.ts
- Update useUser.ts to use GetPreferenceResponse type
- Rename ThemedyProvider to ThemeSyncProvider

dmitri-saricev-3pillargloball committed:
Use GetPreferenceResponse in getUserPreferences

dmitri-saricev-3pillargloball committed:
Merge branch 'v3' into PMM-14624-Fix-theme-inheritance
